### ISSUE TITLE
rpc::client::async_call() blocks if server is unreachable

### ISSUE THREAD

**Issue by bengimizrahi** [rpclib/rpclib#186]
It seems like the function will block in wait_conn() if the server is unreachable. Is there a way to escape from blocking if the server is unreachable? Or am I missing something?

**Comment by bengimizrahi** [comment]
Ok, seems like I have to first check if the client is connected or set a timeout. But I believe users of async_call() would expect the function to be non-blocking even if the connection has not been established.

**Comment by sztomi** [comment]
Yes, that's a very good point, thanks for bringing it forward. I'll fix this in 3.0.
